frameworks/moveos-stdlib/sources/result.move

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) RoochNetwork
+// SPDX-License-Identifier: Apache-2.0
+
+module moveos_std::result {
+
+    use std::string::{Self, String};
+    use std::option::{Self, Option};
+
+    /// Expected the result is ok but the result is err.
+    const ErrorExpectOk: u64 = 1;
+    /// Expected the result is err but the result is ok.
+    const ErrorExpectErr: u64 = 2;
+
+    /// The same as Rust's Result type.
+    /// Most of the time, we do not need the Result type in smart contract, we can directly abort the transaction.
+    /// But in some cases, we need to return a result to ensure the caller can handle the error.
+    struct Result<T> has copy, drop{
+        value: Option<T>,
+        err: Option<String>,
```

Comment:
Can this err also be generic, so that it can support Error types with cause later?

Reply:
I migrated the Result to Nursery and will refractor it in the new PR.